### PR TITLE
Fixes a bug with fireman carry.

### PR DESCRIPTION
**Pull request by uomo91**
## About The Pull Request

PRing this at request of @Rohesie because they're having issues switching their fork over.

## Why It's Good For The Game

Rohesie states: "This fixes a fireman carry bug where you can get tk-like powers if you grab one while standing on a dense object, like a table."

## Changelog
:cl:
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
It's an ugly fix, but the whole feature is brittle as hell. Ideally the component is refactored to avoid the buckle override, but for now it is what it is, and addresses the exploit.

I did the code, btw, not @uomo91 , so yell at me on how evil the patch is.

**Comment by ArcaneDefence** [comment]
This doesn't cover situations where the carrying individual is standing on top of a hand tele's portal, as the person being carried will be teleported to the other side, and the carrier will have the off-hand placeholder.

**Comment by Rohesie** [comment]
Fantastic, brittle code all around. I'll address that.

**Comment by ExcessiveUseOfCobblestone** [comment]
bruh moment

**Comment by ExcessiveUseOfCobblestone** [comment]
sanity checks on the ridingitem vs. having to patch every hole and hoping no one creates anymore (or others existing like portal) seems like a safer option

**Comment by Rohesie** [comment]
The safer option is to nuke this component and refactor it to make it register on an early buckling signal instead of redefining it for humans and the assuming it won't fail later on. But that also means completely re-writing it.

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1004,8 +1004,16 @@
...
 			//Second check to make sure they're still valid to be carried
 			if(can_be_firemanned(target) && !incapacitated(FALSE, TRUE) && !target.buckled)
-				buckle_mob(target, TRUE, TRUE, 90, 1, 0)
-				return
+				if(target.loc != loc)
+					var/old_density = density
+					density = FALSE
+					var/can_step = step_towards(target, loc)
+					density = old_density
+					if(can_step)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm:1012]:
```suggestion
					step_towards(target, loc)
					density = old_density
					if(target.loc == loc)
```

**Pull request merged**